### PR TITLE
[KOGITO-9454] Adding Java embedded flows to examples

### PR DESCRIPTION
**Pull request by fjtirado**
Add embedded sonata workflow examples to be referenced from guide

### CODE REVIEW

File: serverless-workflow-examples/sonata-workflow-fluent/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        <version>${kogito.bom.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+  <dependencies>
+    <dependency>
+       <groupId>org.kie.kogito</groupId>
+       <artifactId>kogito-serverless-workflow-executor</artifactId>
```
Comment (wmedvede) [inline serverless-workflow-examples/sonata-workflow-fluent/pom.xml:31]: @fjtirado  @ricardozanini 

Comment outside this PR. If this executor will be exposed as a kind of entry point to a complementary way to execute sonataflow workflows, maybe we should consider renaming to something like, 

org.apache.kie.sonataflow
sonataflow-worfklow-executor

Or mabye, make the already existing sonataflow-quarkus already include this feature out of the box.

¿?

**Review by wmedvede — APPROVED** [review]

File: serverless-workflow-examples/sonata-workflow-fluent/pom.xml (hunk 1 above)
Comment (fjtirado) [reply serverless-workflow-examples/sonata-workflow-fluent/pom.xml:31]: @wmedvede yes, we will need to eventually rename these dependencies as you propose, sonata-workflow-executor. 
See https://github.com/apache/incubator-kie-kogito-docs/pull/585#discussion_r1525258978

File: serverless-workflow-examples/sonata-workflow-fluent/src/main/java/org/kie/kogito/serverless/workflow/examples/ParallelRest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                                // we are only interested in country name, longitude and latitude
+                                .resultFilter("{country: {name:.[].name.common, latitude: .[].latlng[0], longitude: .[].latlng[1] }}"))
+                        // return only country field to parent flow
+                        .outputFilter("{country}"))
+                .end().build();
+
+        Process<JsonNodeModel> subprocess = application.process(subflow);
+        // This is the main flow, it invokes two services (one for retrieving the age and another to get the gender of the given name )and one subprocess (the country one defined above) in parallel
+        // Once the three of them has been executed, if age is greater than 50, it retrieve the weather information for the retrieved country,
+        // Else, it gets the list of universities for that country.
```
Comment (gmunozfe) [inline serverless-workflow-examples/sonata-workflow-fluent/src/main/java/org/kie/kogito/serverless/workflow/examples/ParallelRest.java:81]: I will express it the other way around (same as in algorithm): if age is less than 50.... else .... (though they are equivalent, it's easy to understand)

**Review by gmunozfe — APPROVED** [review]
Looks good to me, a minor comment, up to you to rephrase it. Nicely done @fjtirado !

File: serverless-workflow-examples/sonata-workflow-fluent/pom.xml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.kie.kogito.examples</groupId>
+    <artifactId>serverless-workflow-examples</artifactId>
+    <version>999-SNAPSHOT</version>
+  </parent>
+  <artifactId>sonata-workflow-fluent</artifactId>
+  <name>Kogito Example :: Sonata Workflow:: Java Embedded examples</name>
```
Comment (ricardozanini) [inline serverless-workflow-examples/sonata-workflow-fluent/pom.xml]:
```suggestion
  <name>Kogito Example :: SonataFlow :: Java Embedded examples</name>
```

File: serverless-workflow-examples/sonata-workflow-fluent/pom.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  <properties>
+    <kogito.bom.group-id>org.kie.kogito</kogito.bom.group-id>
+    <kogito.bom.artifact-id>kogito-bom</kogito.bom.artifact-id>
+    <kogito.bom.version>999-SNAPSHOT</kogito.bom.version>
+    <maven.compiler.release>17</maven.compiler.release>
+    <java.module.name>SonataWorkflowFluent</java.module.name>
```
Comment (ricardozanini) [inline serverless-workflow-examples/sonata-workflow-fluent/pom.xml]:
```suggestion
    <java.module.name>SonataFlowFluent</java.module.name>
```

File: serverless-workflow-examples/sonata-workflow-fluent/pom.xml (hunk 1 above)
Comment (ricardozanini) [reply serverless-workflow-examples/sonata-workflow-fluent/pom.xml:31]: Can we do the renaming now?

**Review by ricardozanini — APPROVED** [review]
We also need a way to create a SonataFlow Fluent Quarkus Extension 🤔 

And have the context as a Quarkus Injectable interface. It would be nice to showcase something like this later. wdyt?

File: serverless-workflow-examples/sonata-workflow-fluent/pom.xml (hunk 1 above)
Comment (fjtirado) [reply serverless-workflow-examples/sonata-workflow-fluent/pom.xml:31]: @ricardozanini I have opened https://github.com/apache/incubator-kie-kogito-runtimes/issues/3447 for the renaming

**Comment by kie-ci3** [comment]
**PR job** `#11` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-examples -u https://github.com/apache/incubator-kie-kogito-examples/pull/1895 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1895/11/display/redirect

**Test results:**
- PASSED: 392
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1895/11/testReport/org.acme/AcmeExchangeResourceIT/unauthorizedTest/">org.acme.AcmeExchangeResourceIT.unauthorizedTest</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor#startKeycloakContainer threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:22.0.5<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:248)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:568)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:864)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:282)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)<br/> at java.base/java.lang.Thread.run(Thread.java:833)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:22.0.5<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:349)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.lambda$startContainer$4(KeycloakDevServicesProcessor.java:374)<br/> at java.base/java.util.Optional.orElseGet(Optional.java:364)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startContainer(KeycloakDevServicesProcessor.java:399)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:196)<br/> ... 11 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)<br/> ... 16 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:553)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 17 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Keycloak.*started.*'<br/> at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)<br/> at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)<br/> at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:964)<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:490)<br/> ... 19 more<br/>
</details>

**Pull request merged**